test/fork/integration/Oracle.t.sol

```diff
@@ -0,0 +1,188 @@
+            vm.warp(block.timestamp + frameStartWithOffset - block.timestamp);
+        }
+    }
+
+    function prepareReport(
+        bytes32 feesTreeRoot,
+        uint256 distributedShares,
+        bytes32 strikesTreeRoot
+    ) public returns (ICSFeeOracle.ReportData memory data) {
+        uint256 consensusVersion = oracle.getConsensusVersion();
+        waitForNextRefSlot();
+        (uint256 refSlot, ) = hashConsensus.getCurrentFrame();
+
+        data = ICSFeeOracle.ReportData({
+            consensusVersion: consensusVersion,
+            refSlot: refSlot,
+            treeRoot: feesTreeRoot,
+            treeCid: "0x1",
+            logCid: "0x2",
+            distributed: distributedShares,
+            rebate: 0,
+            strikesTreeRoot: strikesTreeRoot,
+            strikesTreeCid: "0x3"
+        });
+        reachConsensus(refSlot, keccak256(abi.encode(data)));
+    }
+
+    function test_reportDistributedFees() public {
+        vm.deal(address(feeDistributor), 1 ether);
+        vm.prank(address(feeDistributor));
+        lido.submit{ value: 1 ether }(address(0));
+        uint256 distributed = feeDistributor.pendingSharesToDistribute();
+        feesTree.pushLeaf(abi.encode(nodeOperatorId, distributed));
+        feesTree.pushLeaf(abi.encode(type(uint64).max, 0));
+
+        uint256[] memory strikesData = new uint256[](1);
+        strikesData[0] = 0;
+        strikesTree.pushLeaf(abi.encode(nodeOperatorId, "0x1", strikesData));
+        strikesTree.pushLeaf(
+            abi.encode(nodeOperatorId + 1, "0x1", strikesData)
+        );
+
+        ICSFeeOracle.ReportData memory data = prepareReport(
+            feesTree.root(),
+            distributed,
+            strikesTree.root()
+        );
+        uint256 contractVersion = oracle.getContractVersion();
+        (address[] memory addresses, ) = hashConsensus.getMembers();
+        vm.startPrank(addresses[0]);
+        vm.startSnapshotGas("CSFeeOracle.submitReportData_fees");
+        oracle.submitReportData(data, contractVersion);
+        vm.stopPrank();
+
+        assertEq(feeDistributor.pendingSharesToDistribute(), 0);
+        assertEq(
+            feeDistributor.getFeesToDistribute(
+                nodeOperatorId,
+                distributed,
+                feesTree.getProof(0)
+            ),
+            distributed - feeDistributor.distributedShares(nodeOperatorId)
+        );
+    }
+
+    function test_reportStrikes() public {
+        uint256 distributed = 0;
+        feesTree.pushLeaf(abi.encode(type(uint64).max, 0));
+        uint256 keyIndex = csm.getNodeOperatorTotalDepositedKeys(
+            nodeOperatorId
+        ) - 1;
+        bytes memory key = csm.getSigningKeys(nodeOperatorId, keyIndex, 1);
+
+        (, uint256 threshold) = parametersRegistry.getStrikesParams(
+            accounting.getBondCurveId(nodeOperatorId)
+        );
+        uint256[] memory strikesData = new uint256[](1);
+        strikesData[0] = threshold;
```

Comment:
It won't be single element most of the time, consider adding multiple elements, 3 to 6.